relay/sources/relayd/src/shared_files.rs

```diff
@@ -119,23 +158,30 @@ pub fn parse_value(key: &str, file: &str) -> Result<String, ()> {
 }
 
 pub fn same_hash_than_in_nodeslist(
-    hash_value: &str,
-    source_uuid: String,
-    job_config: Arc<JobConfig>,
-) -> bool {
-    let myvec2: Vec<String> = job_config
-        .nodes
-        .read()
-        .expect("Cannot read nodes list")
-        .get_keyhash_from_uuid(&source_uuid)
-        .unwrap()
-        .split(':')
-        .map(|s| s.to_string())
-        .collect();
+    pubkey: PKey<Public>,
+    hash_type: HashType,
+    keyhash: String,
+) -> Result<bool, Error> {
+    let public_key_der: &[u8] = &pubkey.public_key_to_der()?;
 
-    let (_algorithm, keyhash) = (&myvec2[0], &myvec2[1]);
+    match hash_type {
+        HashType::Sha256 => Ok(HashType::Sha256.hash(public_key_der) == keyhash),
+        HashType::Sha512 => Ok(HashType::Sha512.hash(public_key_der) == keyhash),
+    }
+}
 
-    hash_value == keyhash
+pub fn get_pubkey(
+    pubkey: String,
```

Comment:
nitpick, but it should take `&str` instead of `String` as it does not consume it.